src/App.tsx

```diff
@@ -15,48 +16,117 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const RESET_VALUE = 'Reset';
+const REVERSE_VALUE = 'Reverse';
+
+export enum SortType {
+  Default = '',
+  Alphabeticaly = 'Sort alphabetically',
+  By_Length = 'Sort by length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortKey: SortType,
+  isReversed: boolean,
+) {
+  const goodsCopy = [...goods];
+
+  if (sortKey) {
+    goodsCopy.sort((goodA, goodB) => {
+      switch (sortKey) {
+        case SortType.Alphabeticaly:
+          return goodA.localeCompare(goodB);
+
+        case SortType.By_Length:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    goodsCopy.reverse();
+  }
+
+  return goodsCopy;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleResetButton = sortType || isReversed;
```

Comment:
The name can't start with the word `handle` because it is not a function
As it has `boolean` value we should start name with `is`